applications/StructuralMechanicsApplication/custom_processes/set_moving_load_process.cpp

```diff
@@ -0,0 +1,319 @@
+
+ 
+    while (visited_indices.size() != unsorted_conditions_v.size())
+    {
+        for (int i =0; i< unsorted_conditions_v.size(); i++)
+        {
+            Condition& r_cond = unsorted_conditions_v[i];
+            GeometricalObject::GeometryType& r_geom = r_cond.GetGeometry();
+
+
+            // check if current index is already added to sorted condition vector
+            if (!std::count(visited_indices.begin(), visited_indices.end(), i))
+            {
+                // check if geom has a shared node with previous geom
+                if (node_id_vector.find(r_geom.Points()[0].Id()) != node_id_vector.end() || node_id_vector.find(r_geom.Points()[1].Id()) != node_id_vector.end())
+                {
+                    if (sorted_conditions.size() == 0)
+                    {
+                        // check if both nodes of geom are equal to nodes in start element, only do this to add the first element in the sorted conditions vector
+                        if (node_id_vector.find(r_geom[0].Id()) != node_id_vector.end() && node_id_vector.find(r_geom.Points()[1].Id()) != node_id_vector.end())
+                        {
+                            node_id_vector = { r_geom[0].Id(),r_geom[1].Id() };
+                            sorted_conditions.push_back(r_cond);
+                            visited_indices.push_back(i);
+                        }
+                    }
+                    else
+                    {
+                        node_id_vector = { r_geom[0].Id(),r_geom[1].Id() };
+                        sorted_conditions.push_back(r_cond);
+                        visited_indices.push_back(i);
+                    }
+                }
+            }
+            
+        }
+    }
+
+    // swap first 2 conditions if nessesary
+    //r_geom = first_condition.GetGeometry();
+    int tmp = first_condition.Id();
+    for (int i = 0; i < sorted_conditions.size(); ++i)
+    {
+        int tmp2 = sorted_conditions[i].Id();
+        double tmp12 = sorted_conditions[i].GetGeometry()[0].X();
+        int tmp7 = 2;
+    }
+    
+
+    return sorted_conditions;
+}
+
+
+void SetMovingLoadProcess::ExecuteInitialize()
+{
+    KRATOS_TRY
+    mLoad = mParameters["load"].GetVector();
+    const double current_time = mrModelPart.GetProcessInfo().GetValue(TIME);
+
+    vector<int> direction = mParameters["direction"].GetVector();
+    mLoadVelocity = mParameters["velocity"].GetDouble();
+    mCurrentDistance = 0;
+
+    std::vector<int> node_id_vector;
+    std::vector<int> node_id_vector2;
+
+    // get all end node ids ( not the middle nodes, in case of line3 conditions)
+    for (auto& r_cond : mrModelPart.Conditions()) {
+
+        auto geom = r_cond.GetGeometry();
+        node_id_vector.push_back(geom[0].Id());
+        node_id_vector.push_back(geom[1].Id());
+    }
+
+    // find non repeating node ids
+    std::vector<int> non_repeating_node_ids = FindNonRepeatingIndices(node_id_vector);
```

Comment:
I think you can easily achieve the same with an `unordered_set`.

Reply:
thanks, I felt like there was a better way than I was doing

Reply:
actually, I can not use unordered_set, as my intention is to find the node id's which are not duplicated. I don't want to just find the unique id's

Reply:
OK. Now I see your point, you need the "init and end points" id's. Maybe you can store all the ids in a vector, sort them and use `std::unique`.

Reply:
If I have 10 line2D2 condition elements, I get 20 points. Using std::unique results in 11 unique points. What i want is to only find the very first and last node id, so 2 points. 

I also do a check if 2 points are found. If more than 2 points are found, it means that some conditions are not connected, thus i raise an error.

Reply:
True. `std::unique` keeps the first of the repeated entries... Then nothing comes to my mind.